### PR TITLE
Lobby speedup

### PR DESCRIPTION
**Pull request by Crotalus**
A feeble attempt at reducing some lobby lag 

* Only send game rules to the joining player
* Send benchmark results in one batch
* Support for sending multiple game options 
* Support for sending multiple player options
* Cache result of CPU benchmark

Some more info in the commit messages

### CODE REVIEW

**Comment by ChrisKitching** [comment]
These are, in general, good ideas.

The reasons I didn't do these already are basically:
- Since these involve the wire protocol, they're mildly irritating to test.
- Because of that, there's scope for breakage if you get this wrong.
- Optimising network messages isn't that useful for reducing UI lag. Cutting down the calls to UpdateGame (and eventually killing it entirely) will make a difference (you're doing some of that here, too, which is encouraging).

In short: This looks generally great, but needs very careful and comprehensive testing, and don't get too excited about the perf wins you'll see.

**Comment by Crotalus** [comment]
Good input! I've updated the PR now. 

Yeah, some of these changes helps with reducing # calls to UpdateGame() so hopefully some improvement.

**Comment by Sheeo** [comment]
We'll need some thorough testing of this with the server, making sure it doesn't act up.

**Comment by ChrisKitching** [comment]
@Crotalus We've decided to go nuts and land the various bits of "make the lobby comm protocol not stupid" work that've been lying around.
If you could rebase this on current develop (which I just set fire to), you're welcome to merge it right away.
Don't forget to throw away the very last commit: I merged that earlier on.

**Comment by Crotalus** [comment]
Removed the rule stuff also, saw that you use GameOption for that now
